app/src/views/private/components/overlay-item.vue

```diff
@@ -0,0 +1,625 @@
+			if (!isGroup) return;
+
+			if (isSelected(field.field)) groupKeys.push(field.field);
+		});
+
+		return addNestedGroupKeys(groupKeys);
+	}
+
+	function addNestedGroupKeys(groupKeys: string[]) {
+		availableFields.forEach((field) => {
+			const isGroup = field.meta?.special?.includes('group');
+			if (!isGroup) return;
+
+			const groupIsAlreadySelected = groupKeys.includes(field.field);
+			if (groupIsAlreadySelected) return;
+
+			const groupKey = field.meta?.group;
+			if (!groupKey) return;
+
+			const parentGroupIsSelected = groupKeys.includes(groupKey);
+			if (!parentGroupIsSelected) return;
+
+			groupKeys.push(field.field);
+			groupKeys = addNestedGroupKeys(groupKeys);
+		});
+
+		return groupKeys;
+	}
+});
+
+const fieldsWithoutCircular = computed(() => {
+	if (props.circularField) {
+		return fields.value.filter((field) => {
+			return field.field !== props.circularField;
+		});
+	} else {
+		return fields.value;
+	}
+});
+
+const templatePrimaryKey = computed(() =>
+	junctionFieldInfo.value ? String(props.relatedPrimaryKey) : String(props.primaryKey),
+);
+
+const templateCollection = computed(() => relatedCollectionInfo.value || collectionInfo.value);
+
+const isSavable = computed(() => {
+	if (props.disabled) return false;
+	if (!relatedCollection.value) return saveAllowed.value;
+	return saveAllowed.value || saveRelatedCollectionAllowed.value;
+});
+
+const {
+	template,
+	templateData,
+	loading: templateDataLoading,
+} = useTemplateData(templateCollection, templatePrimaryKey);
+
+function useActiveState() {
+	const localActive = ref(false);
+
+	const internalActive = computed({
+		get() {
+			return props.active ?? localActive.value;
+		},
+		set(newActive: boolean) {
+			localActive.value = newActive;
+			emit('update:active', newActive);
+		},
+	});
+
+	return { internalActive };
+}
+
+function useItem() {
+	const internalEdits = ref<Record<string, any>>({});
+	const loading = ref(false);
+	const initialValues = ref<Record<string, any> | null>(null);
+
+	watch(
+		() => props.active,
+		(isActive) => {
+			if (isActive) {
+				if (props.primaryKey !== '+') fetchItem();
+				if (props.relatedPrimaryKey !== '+') fetchRelatedItem();
+				internalEdits.value = props.edits ?? {};
+			} else {
+				loading.value = false;
+				initialValues.value = null;
+				internalEdits.value = {};
+			}
+		},
+		{ immediate: true },
+	);
+
+	return { internalEdits, loading, initialValues, refresh };
+
+	async function refresh() {
+		if (props.active) {
+			loading.value = false;
+			if (props.primaryKey !== '+') fetchItem();
+			if (props.relatedPrimaryKey !== '+') fetchRelatedItem();
+		}
+	}
+
+	async function fetchItem() {
+		if (!props.primaryKey) return;
+
+		loading.value = true;
+
+		const baseEndpoint = getEndpoint(props.collection);
+
+		const endpoint = isSystemCollection(props.collection)
+			? `${baseEndpoint}/${props.primaryKey}`
+			: `${baseEndpoint}/${encodeURIComponent(props.primaryKey)}`;
```

Comment:
Could extract this logic to a `getCombinedEndpoint(collection, key)` function as it's the same except for the props here and below